pkg/batchpertrace/batchpertrace_test.go

```diff
@@ -56,3 +56,39 @@ func TestSplitDifferentTracesIntoDifferentBatches(t *testing.T) {
 	assert.Equal(t, library.Name(), secondOutILS.InstrumentationLibrary().Name())
 	assert.Equal(t, secondSpan.Name(), secondOutILS.Spans().At(0).Name())
 }
+
+func TestSplitDifferentLogsIntoDifferentBatches(t *testing.T) {
+	// we have 1 ResourceSpans with 1 ILS and two traceIDs, resulting in two batches
+	inBatch := pdata.NewLogs()
+	inBatch.ResourceLogs().Resize(1)
+	rl := inBatch.ResourceLogs().At(0)
+	rl.InstrumentationLibraryLogs().Resize(1)
+
+	// the first ILS has two spans
+	ill := rl.InstrumentationLibraryLogs().At(0)
+	library := ill.InstrumentationLibrary()
+	library.SetName("first-library")
+	ill.Logs().Resize(2)
+	firstLog := ill.Logs().At(0)
+	firstLog.SetName("first-batch-first-log")
+	firstLog.SetTraceID(pdata.NewTraceID([16]byte{1, 2, 3, 4}))
```

Comment:
Could you add a test for a log entry that doesn't have a trace ID?

Reply:
Done here: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/2549/files#diff-5b927bd0ee2bf7e5ee21db9bf4a4a6e4406b4f75e83ff616c5d176117ef852d5R80